pkg/agent/globalstate.go

```diff
@@ -153,8 +153,11 @@ func (s *agentState) handleVMEventAdded(
 			state:              "", // Explicitly set state to empty so that the initial state update does no decrement
 			stateUpdatedAt:     now,
 
-			startTime:                 now,
-			lastSuccessfulMonitorComm: nil,
+			startTime:                     now,
+			lastSuccessfulMonitorComm:     nil,
+			failedMonitorRequestCounter:   util.NewRecentCounter(time.Duration(s.config.Monitor.MaxFailedRequestRate.IntervalSeconds) * time.Second),
+			failedNeonVMRequestCounter:    util.NewRecentCounter(time.Duration(s.config.NeonVM.MaxFailedRequestRate.IntervalSeconds) * time.Second),
+			failedSchedulerRequestCounter: util.NewRecentCounter(time.Duration(s.config.Scheduler.MaxFailedRequestRate.IntervalSeconds) * time.Second),
```

Comment:
I guess this is a case where #35 would help (specifically: so we don't need to do the verbose conversion from integer seconds to `time.Duration`)